downstairs/src/region.rs

```diff
@@ -32,8 +32,30 @@ use crate::extent::{
  * There is only one repair file: the raw file itself (which also contains
  * structured context and metadata at the end).
  */
-pub fn validate_repair_files(eid: usize, files: &[String]) -> bool {
-    files == [extent_file_name(eid as u32, ExtentType::Data)]
+pub fn validate_repair_files(
+    eid: usize,
+    files: &[String],
+    replacement: bool,
+) -> bool {
+    if replacement {
+        let one = vec![extent_file_name(eid as u32, ExtentType::Data)];
+
+        let mut some = one.clone();
+        some.extend(vec![extent_file_name(eid as u32, ExtentType::Db)]);
+
+        let mut all = some.clone();
+        all.extend(vec![
+            extent_file_name(eid as u32, ExtentType::DbShm),
+            extent_file_name(eid as u32, ExtentType::DbWal),
+        ]);
```

Comment:
Are these always guaranteed to be in the same order?  Using a `BTreeSet<String>` or similar would make it more obvious that the comparison is meant to be order-agnostic.

Reply:
We always produce them in the same order, so assuming `extent_file_list()`'s `into_iter()` call
will walk the `vec` provided to it in order, then I suppose it's safe to expect that order never
changes?  
We also (and I'm not sure why any longer) call `repair_files.sort()` over in
`downstairs/src/region.rs`, so even if the remote side does not produce the files in order, we
sort them on the requesting side.

Now that I think about it, I wonder if that `repair_files.sort()` in downstairs/src/region.rs was
attempting to work around the bug in the file type translation, where we had mixed `db-shm`
and `db-wal`.